first-party/jib-layer-filter-extension-maven/src/main/java/com/google/cloud/tools/jib/maven/extension/layerfilter/JibLayerFilterExtension.java

```diff
@@ -102,9 +122,120 @@ public ContainerBuildPlan extendContainerBuildPlan(
         .map(FileEntriesLayer.Builder::build)
         .filter(layer -> !layer.getEntries().isEmpty())
         .forEach(newPlanBuilder::addLayer);
+
+    ContainerBuildPlan newPlan = newPlanBuilder.build();
+
+    if (config.get().isCreateParentLayers()) {
+      newPlan = moveParentDepsToNewLayers(newPlan, mavenData, logger);
+    }
+
+    return newPlan;
+  }
+
+  private ContainerBuildPlan moveParentDepsToNewLayers(
+      ContainerBuildPlan buildPlan, MavenData mavenData, ExtensionLogger logger)
+      throws JibPluginExtensionException {
+
+    logger.log(LogLevel.LIFECYCLE, "Moving parent dependencies to new layers.");
+
+    // the key is the expected path for the parent dependency
+    Map<String, Artifact> parentDependencies =
+        getParentDependencies(mavenData)
+            .stream()
+            .map(d -> d.getArtifact())
+            // TODO: configurable?
+            .collect(
+                // TODO: getVersion or getBaseVersion? check in jib, which version it is...
+                Collectors.toMap(
+                    a -> "/app/libs/" + a.getArtifactId() + "-" + a.getVersion() + ".jar", a -> a));
+
+    // parent dependencies that have not been found in any layer (due to different version or
+    // filtering)
+    Map<String, Artifact> parentDependenciesNotFound = new HashMap<>(parentDependencies);
```

Comment:
Looks like this doesn't have to be a map.
```suggestion
    Set<String> missingParentDependencies = parentDependencies.keySet();
```

Reply:
Will leave it open until I have added the code that actually uses the map.

Reply:
with https://github.com/GoogleContainerTools/jib-extensions/pull/76/commits/1a5d3210c2aaedad2e96401d48d1c22cfc7404cd I now use the artifact from the map to find and log potential matches.